[docs] Update control-plane-manager architecture section in accordance with PR 17798 (#20390)
deckhouse/deckhouse: docs/documentation/pages/architecture/kubernetes-and-scheduling/CONTROL_PLANE_MANAGEMENT.md

Cluster control plane components are managed by the [`control-plane-manager`](/modules/control-plane-manager/) module, which runs on all master nodes (nodes labeled with `node-role.kubernetes.io/control-plane: ""`).

The module operates with the following custom resources:

> - [ControlPlaneNode](/modules/control-plane-manager/cr.html#controlplanenode): Describes the parameters and state of control plane nodes (master nodes) in the cluster. Its used for managing the lifecycle and configuration of each control plane component.

Reviewer: ```suggestion
The module operates with the following custom resources:

- [ControlPlaneNode](/modules/control-plane-manager/cr.html#controlplanenode): Describes the parameters and state of control plane nodes (master nodes) in the cluster. It's used for managing the lifecycle and configuration of each control plane component.
```

The module consists of the following components:

1. **d8-control-plane-manager** (DaemonSet): Manages cluster control plane components and runs on all master nodes.

   The **d8-control-plane-manager** controller performs the following actions:

      - Monitors the `d8-control-plane-manager-config` and `d8-pki` Secrets and, based on their information, creates or updates the ControlPlaneNode custom resource for each master node.

      - If the required node configuration differs from the current one, creates a ControlPlaneOperation resource to perform operations to update the node's configuration.

      - Determines the order in which to execute the requested ControlPlaneOperation operations to maintain the required cluster fault tolerance during updates.

      - Monitors the execution of operations specified in the ControlPlaneOperation resource.

>       - After the requested operations are completed, updates the current state of the master node in the ControlPlaneNode resource.

Reviewer: ```suggestion
   - Monitors the `d8-control-plane-manager-config` and `d8-pki` Secrets and, based on their information, creates or updates the ControlPlaneNode custom resource for each master node.

   - If the required node configuration differs from the current one, creates a ControlPlaneOperation resource to perform operations to update the node's configuration.

   - Determines the order in which to execute the requested ControlPlaneOperation operations to maintain the required cluster fault tolerance during updates.

   - Monitors the execution of operations specified in the ControlPlaneOperation resource.

   - After the requested operations are completed, updates the current state of the master node in the ControlPlaneNode resource.
```

1. **kube-apiserver**:

   * Manages cluster control plane components.
   * Reconciles ControlPlaneNode and ControlPlaneOperation custom resources.

>    * Watches Secret resources `d8-control-plane-manager-config` and `d8-pki`.

Reviewer 2: ```suggestion
   * Watches `d8-control-plane-manager-config` and `d8-pki` Secrets.
```
